src/main/java/com/iexec/worker/executor/TaskExecutorService.java

```diff
@@ -251,22 +280,41 @@ public void abortContributionTimeout(String chainTaskId) {
     public void uploadResult(String chainTaskId) {
         customFeignClient.updateReplicateStatus(chainTaskId, RESULT_UPLOADING);
 
-        Eip712Challenge eip712Challenge = resultRepoClient.getChallenge(publicConfigurationService.getChainId());
+        boolean isResultEncrypted = resultService.encryptResult(chainTaskId);
```

Comment:
Do we need to encrypt the result all the time?

Reply:
I added a condition to decrypt only when needed